dbms/src/Common/ProfilingScopedRWLock.h

```diff
@@ -17,6 +17,17 @@ class ProfilingScopedWriteRWLock
         ProfileEvents::increment(event, watch.elapsed());
     }
 
+    /// Unsafe!
```

Comment:
The comment is not helpful, we should explain more or remove it. Looks like this is just a generic lock/unlock method that is not particularly unsafe.

Reply:
Also we should increment the metric when we re-lock.